drivers/SmartThings/matter-thermostat/src/init.lua

```diff
@@ -214,10 +245,86 @@ local function set_setpoint(setpoint)
       value = f_to_c(value)
     end
 
+    -- Gather cached setpoint values when considering setpoint limits
+    -- Note cached values should always exists, but defaults are chosen just in case to prevent
+    -- nil operation errors, and deadband logic from triggering.
+    local cooling_setpoint = device.state_cache.main.thermostatCoolingSetpoint.coolingSetpoint or { value = 100 }
+    if cooling_setpoint and cooling_setpoint.unit == "F"then
+      cooling_setpoint = f_to_c(cooling_setpoint.value)
```

Comment:
unless full precision is desired, you can probably switch this over to the `util` function now, which gives two digits of precision

Reply:
Looks like that change is in the 0.45 release, so I will make it.